lint/util.py

```diff
@@ -44,18 +44,20 @@ def printf(*args):
 
 
 def get_syntax(view):
-    """
-    Return the view's syntax.
-
-    or the syntax it is mapped to in the "syntax_map" setting.
-    """
+    """Return the view's syntax."""
     from .persist import settings
     syntax_map = settings.get('syntax_map')
-    for syntax, selectors in syntax_map.items():
-        for s in selectors:
-            if view.match_selector(0, s):
-                return syntax
-    return ""
+    try:
+        scoring = {
+            max(view.score_selector(0, s) for s in selectors if s): syntax
+            for syntax, selectors in syntax_map.items()
+        }
+        max_score = max(scoring, key=scoring.get)
```

Comment:
Pretty sure the provided `key` is doing the opposite of what you want here. It will order the scores after the associated syntax string and not by their numeric value.

Reply:
Good spot. I have switched to score: syntax.
That handles the unlikely case we get two syntaxes with the same score.
The latter in our list overwriting the former.

Reply:
@FichteFoll based on the current state of code. Could you make a suggestion how to implement that fallback mechanism? Is it as simple as, splitting - the - scope on whitespace and iterating the scoring algorithm over its fragments? What am I missing here?

Reply:
Pretty much? I mean, I linked a reference implementation.